core/src/main/java/org/polypheny/db/ResultIterator.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2019-2023 The Polypheny Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.polypheny.db;
+
+import static org.reflections.Reflections.log;
+
+import com.google.common.collect.Lists;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.StreamSupport;
+import javax.annotation.Nullable;
+import lombok.Value;
+import org.apache.commons.lang3.time.StopWatch;
+import org.jetbrains.annotations.NotNull;
+import org.polypheny.db.algebra.type.AlgDataType;
+import org.polypheny.db.catalog.exceptions.GenericRuntimeException;
+import org.polypheny.db.monitoring.events.StatementEvent;
+import org.polypheny.db.routing.ExecutionTimeMonitor;
+import org.polypheny.db.transaction.Statement;
+import org.polypheny.db.type.entity.PolyValue;
+
+
+@Value
+public class ResultIterator implements AutoCloseable {
+
+    Iterator<PolyValue[]>
+            iterator;
+    int batch;
+    ExecutionTimeMonitor executionTimeMonitor;
+    boolean isIndex;
+    boolean isTimed;
+    AlgDataType rowType;
+    StatementEvent statementEvent;
+    PolyImplementation implementation;
```

Comment:
Can these be declared final and have a lower access level?

Reply:
Class is sealed with @Value/@Data.